CRM/Contribute/Form/Task/Invoice.php

```diff
@@ -543,7 +543,11 @@ public static function printPDF($contribIDs, &$params, $contactIds) {
         self::addActivities($subject, $contribution->contact_id, $fileName, $params);
       }
 
-      CRM_Core_DAO::setFieldValue('CRM_Contribute_DAO_Contribution', $contribution->id, 'invoice_id', $invoiceId);
+      // CRM-20387 save my human-readable invoice number as invoice_id in the contribution table,
+      // but only if the current value is empty
+      if (empty($contribution->invoice_id)) {
```

Comment:
This will cause a regression for organizations that need to comply with UK VAT rules.